compiler/noirc_frontend/src/elaborator/patterns.rs

```diff
@@ -1004,6 +1004,22 @@ impl Elaborator<'_> {
         let (typ, bindings) =
             self.instantiate(t, bindings, generics, function_generic_count, location);
 
+        if let ImplKind::TraitItem(mut method) = ident.impl_kind {
```

Comment:
Can you add a comment either above this if or above line 1025 explaining why this constraint needs to be pushed before any subsequent ones?

Reply:
Done! I used the same code for the example to try to explain it. Let me know if it's fine. I feel like it could be explained better, but I'm not sure how.

In any case I think because of the new regression test the order can't be reverted back.